### PR TITLE
feat: improve abstraction for external providers

### LINKED ISSUE #114: feat: support for local models (ollama)
Seeing how easy fabrice makes it to spin up a local agent CLI, it would be neat to hook it up with local models like https://github.com/ollama/ollama

I'm sure more people would like to test fabrice in a professional setting, but can't process company data on external models (i.e.: their own OpenAI API keys).

### PR DESCRIPTION
**Pull request by grabbou**
Slightly better abstraction (but still light-weight) for other providers. 
Support for Ollama (tho results with local LLMs are rather poor, I would love if someone could experiment with it).
We may need to tweak prompts on per model basis, which I would see as a task for the future.

Fixes #114

### CODE REVIEW

File: packages/framework/src/providers/ollama.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      .optional(),
+  }),
+  done_reason: z.string(),
+  done: z.boolean(),
+  total_duration: z.number(),
+  load_duration: z.number(),
+  prompt_eval_count: z.number(),
+  prompt_eval_duration: z.number(),
+  eval_count: z.number(),
+  eval_duration: z.number(),
```
Comment (pkarw) [inline packages/framework/src/providers/ollama.ts:37]: Shouldn't we have the tokens usage here? I think it's the default w OpenAI response (?)

File: packages/framework/src/providers/ollama.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              arguments: JSON.stringify(toolCall.function.arguments),
+              parsed_arguments: toolCall.function.arguments,
+            },
+          })) || [],
+        refusal: null,
+        parsed,
+      }
+    },
+
+    embeddings: async (input: string) => {
```
Comment (pkarw) [inline packages/framework/src/providers/ollama.ts:95]: That's great you moved this one too to provider 🙌

**Review by pkarw — APPROVED** [review]

**Comment by pkarw** [comment]
Great job. Approved with just one question regarding the token usage stats within LLM response - if not now we probably will add it later nevertheless, I mean it's kind of standard per LLM responsy in any framework

**Comment by grabbou** [comment]
Right now, Ollama won't work due to https://github.com/ollama/ollama/issues/8095, unless I misconfigured the request. I will try to replicate this behavior by tweaking the schema and we will see.

**Review by pkarw — APPROVED** [review]
Approved with two minor suggestions

File: packages/framework/src/providers/grok.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Provider } from '../models.js'
+import { OpenAIOptions } from './openai.js'
```
[the remark this answers is not in the document]
Comment (grabbou) [reply packages/framework/src/providers/grok.ts]: Renamed as OpenAIProviderOptions as these are openai() specific settings, but not to confuse with OpenAI client options!

**Comment by grabbou** [comment]
I am going to merge it, but let's test it thoroughly tomorrow before cutting release, to make sure it does work as expected!

**Comment by grabbou** [comment]
Grok working good!

**Pull request merged**